tests/sycl/sub_group.cpp

```diff
@@ -108,6 +108,13 @@ BOOST_AUTO_TEST_CASE(sub_group) {
       }
     }
   }
+
+  const s::device dev = q.get_device();
+  const std::vector<size_t> supported_subgroup_sizes = dev.get_info<cl::sycl::info::device::sub_group_sizes>();
+  BOOST_CHECK(supported_subgroup_sizes.size() == 1);
+  BOOST_CHECK(supported_subgroup_sizes[0] == subgroup_size);
```

Comment:
The assumptions behind this part of the test has changed recently because we can now target multiple backends simultaneously. In that case we run into trouble with the assumption that we can just determine the expected subgroup size based on those macros. I don't think we can actually test this apart from testing that it compiles.